### PR TITLE
More use of hashmap_ensure_put and ordered_hashmap_ensure_put

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/core/transaction.c [hunk 1]
```diff
@@ -648,11 +648,9 @@ static int transaction_apply(
                 assert(!j->transaction_prev);
                 assert(!j->transaction_next);
 
-                r = hashmap_ensure_allocated(&m->jobs, NULL);
-                if (r < 0)
+                r = hashmap_ensure_put(&m->jobs, NULL, UINT32_TO_PTR(j->id), j);
+                if (r == -ENOMEM)
                         return r;
```
Comment (yuwata) [inline src/core/transaction.c]: Please drop these two lines, and always goto `rollback`.

File: src/network/networkd-radv.c [hunk 2]
```diff
@@ -143,11 +143,9 @@ static int route_prefix_new_static(Network *network, const char *filename, unsig
...
         prefix->section = TAKE_PTR(n);
 
-        r = hashmap_ensure_allocated(&network->route_prefixes_by_section, &network_config_hash_ops);
-        if (r < 0)
-                return r;
-
-        r = hashmap_put(network->route_prefixes_by_section, prefix->section, prefix);
+        r = hashmap_ensure_put(&network->route_prefixes_by_section, &network_config_hash_ops, prefix->section, prefix);
+        if (r == -ENOMEM)
+                return log_oom();
```
Comment (yuwata) [inline src/network/networkd-radv.c]: Please drop these two lines. This function should not log anything.

File: src/remount-fs/remount-fs.c [hunk 3]
```diff
@@ -31,17 +31,15 @@ static int track_pid(Hashmap **h, const char *path, pid_t pid) {
...
         if (!c)
                 return log_oom();
 
-        r = hashmap_put(*h, PID_TO_PTR(pid), c);
-        if (r < 0)
+        r = hashmap_ensure_put(h, NULL, PID_TO_PTR(pid), c);
+        if (r == -ENOMEM)
                 return log_oom();
+        if (r < 0)
+                return r;
```
Comment (yuwata) [inline src/remount-fs/remount-fs.c]: Please add an error log here.
```
if (r < 0)
    return log_error_errno(r, "Failed to store pid " PID_FMT, pid);
```

File: src/sysusers/sysusers.c [hunk 4]
```diff
@@ -1311,22 +1307,20 @@ static int add_implicit(void) {
                                 if (!j->name)
                                         return log_oom();
 
-                                r = ordered_hashmap_put(users, j->name, j);
-                                if (r < 0)
+                                r = ordered_hashmap_ensure_put(&users, &item_hash_ops, j->name, j);
+                                if (r == -ENOMEM)
                                         return log_oom();
+                                if (r < 0)
+                                        return r;
```
Comment (yuwata) [inline src/sysusers/sysusers.c]: Please add error log here.
```
if (r < 0)
    return log_error_errno(r, "Failed to add implicit user '%s': %m", j->name);
```

File: src/sysusers/sysusers.c [hunk 5]
```diff
@@ -1336,12 +1330,14 @@ static int add_implicit(void) {
                         if (!j->name)
                                 return log_oom();
 
-                        r = ordered_hashmap_put(groups, j->name, j);
-                        if (r < 0)
+                        r = ordered_hashmap_ensure_put(&groups, &item_hash_ops, j->name, j);
+                        if (r == -ENOMEM)
                                 return log_oom();
+                        if (r < 0)
+                                return r;
```
Comment (yuwata) [inline src/sysusers/sysusers.c]: Same here.
```
if (r < 0)
    return log_error_errno(r, "Failed to add implicit group '%s': %m", j->name);
```

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/core/transaction.c (hunk 1 above)
Comment (ssahani) [reply src/core/transaction.c]: umm makes sense done

**Comment by ssahani** [comment]
updated thanks for the review ! ptal

**Review by yuwata — APPROVED** [review]
LGTM.

**Pull request merged**